ui/src/app/apps/apps.component.ts

```diff
@@ -86,31 +127,67 @@ export class AppsComponent implements OnInit {
 
         if (this.appsService.appRegistrations.items.length === 0 && this.appsService.appRegistrations.pageNumber > 0) {
           this.appRegistrations.pageNumber = this.appRegistrations.pageNumber - 1;
-          this.loadAppRegistrations(true);
         }
+        const reloadAppsSubscription = this.appsService.getApps(true).subscribe(
+          appRegistrations => {}
+        );
+        this.busy = reloadAppsSubscription;
       },
       error => {
         this.toastyService.error(error);
       }
     );
   }
 
+  /**
+   * Completes the unregistration step for multiple {@link AppRegistration}s.
+   * Closes the open modal dialog.
+   *
+   * @param appRegistrations The array of AppRegistrations to unregister
+   */
   public proceedToUnregisterMultipleAppRegistrations(appRegistrations: AppRegistration[]): void {
     console.log(`Proceeding to unregister ${appRegistrations.length} application(s).`, appRegistrations);
-    for (const appRegistrationToUnregister of appRegistrations) {
-      this.proceedToUnregisterSingleAppRegistration(appRegistrationToUnregister);
-    }
+    const subscription = this.appsService.unregisterMultipleApps(appRegistrations).subscribe(
+      data => {
+        console.log(data);
+        this.toastyService.success(`${data.length} app(s) unregistered.`);
+
+        if (this.appsService.appRegistrations.items.length === 0 && this.appsService.appRegistrations.pageNumber > 0) {
+          this.appRegistrations.pageNumber = this.appRegistrations.pageNumber - 1;
+        }
+        const reloadAppsSubscription = this.appsService.getApps(true).subscribe(
```

Comment:
this.busy can be used instead of reloadAppsSubscription.  This can be applied to the others.